subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/BasicJavaCompilerIntegrationSpec.groovy

```diff
@@ -143,18 +143,18 @@ public class FxApp extends Application {
 
     @Requires(TestPrecondition.JDK9_OR_LATER)
     @Unroll
-    def "compile with release option"() {
+    def "compile with release flag"() {
         given:
         goodCode()
         buildFile << """
-java.targetCompatibility = JavaVersion.VERSION_1_7
+java.targetCompatibility = JavaVersion.VERSION_1_7 // this will be ignored when compiling, but used for the TargetJvmVersion attribute
```

Comment:
this comment makes we wonder if you need this even though you specify the release flag already

Reply:
Also, why would this test use the `--release` flag instead of the new property?

Reply:
That change was a rebase mistake. The test should not have changed, fix incoming.
The test uses `--release` because it illustrates that even when using that, the `targetJvmVersion` attribute has the right value.